lib/resources/postgres.rb

```diff
@@ -16,7 +16,7 @@ def initialize
       # print warnings if the dirs do not exist
       verify_dirs
 
-      if !@version.nil? && !@conf_dir.empty?
+      if !@version.to_s.empty? && !@conf_dir.to_s.empty?
```

Comment:
```ruby
[6] pry(main)> "test".empty?
=> false

[7] pry(main)> "".empty?
=> true

[8] pry(main)> nil.empty?
NoMethodError: undefined method `empty?' for nil:NilClass
from (pry):8:in `__pry__'

[9] pry(main)> nil.to_s.empty?
=> true

[10] pry(main)> "test".to_s.empty?
=> false

[11] pry(main)> "".to_s.empty?
=> true
```

Reply:
Why is `@version.nil?` converted as well?

As far as the second one goes, I think it is better to use a safe-call to preserve meaning (tho, I'd rather we walk upstream of that nil and figure out how to defend against it at the source):

```
>> nil&.empty?
=> nil
>> ""&.empty?
=> true
>> "test"&.empty?
=> false
```

Reply:
Gave the save operator a try but doesn't look to handle the nil as we need it for these conditions.
```ruby
$ pry
[1] pry(main)> @version
=> nil
[2] pry(main)> puts "UNDEFINED" if @version&.empty?
=> nil
[3] pry(main)> puts "UNDEFINED" if @version.to_s.empty?
UNDEFINED
=> nil
[4] pry(main)> @version=''
=> ""
[5] pry(main)> puts "UNDEFINED" if @version&.empty?
UNDEFINED
=> nil
[6] pry(main)> puts "UNDEFINED" if @version.to_s.empty?
UNDEFINED
=> nil
[7] pry(main)> @version='1.2.3'
=> "1.2.3"
[8] pry(main)> puts "UNDEFINED" if @version&.empty?
=> nil
[9] pry(main)> puts "UNDEFINED" if @version.to_s.empty?
=> nil
```

Reply:
Nils or empty strings can come from three methods
```
version_from_psql
version_from_dir
locate_data_dir_location_by_version
```
and possibly more in the future.

I wanted to guard parts of the code vulnerable to empty values to avoid the stacktraces I've got without trying hard.

Reply:
It's the covering it up and moving on part that I'm a bit worried about, but I'm not familiar with the tools being used yet.

Just a coding smell: silently hiding nil in succession is usually a bandaid on a bandaid.